### ISSUE TITLE
on shutdown, kill all children

### ISSUE THREAD

**Issue by janschulz** [IRkernel/IRkernel#307]
theoretically, a user can open subprocesses and these are kept around as orphaned processes because of the way the notebook server kills the kernels. Upon receiving the shutdown request, the kernel should query record all children of itself and kill them and after doing that (and logging any surviving children) kill itself.

**Comment by flying-sheep** [comment]
well, we shouldn’t overreact. sometimes i also feel like killing all children, but many a re kinda nice if you get to know them :joy:

**Comment by janschulz** [comment]
Writing comments in jupyter/jupyter_client#104 was way too much fun... "kernel massacre"-function :-)

**Comment by takluyver** [comment]
Yeah, you have to look out, you can end up on watchlists if you spend too long searching for "how to kill daemon children" ;-)

**Comment by takluyver** [comment]
On a more serious note, do we know of any good API to find all children of the current process?

**Comment by janschulz** [comment]
I just found this, but this states at the beginning that it is not available on windows... https://stat.ethz.ch/R-manual/R-devel/library/parallel/html/children.html
